### PR TITLE
Add `RandomAI` and tests for it

### PR DESCRIPTION
**Pull request by Yo1k**

### CODE REVIEW

File: yo1k/tic_tac_toe/test/ai/test_ai.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+class RandomAITest(unittest.TestCase):
+    def test_play_against_itself(self) -> None:
+        for _ in range(100):
+            ai_rng_seed_px = randrange(sys.maxsize)
+            ai_rng_seed_po = randrange(sys.maxsize)
+            player_x = Player(PlayerID(0), Mark.X)
+            player_o = Player(PlayerID(1), Mark.O)
+            px_id = player_x.id
+            po_id = player_o.id
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/test/ai/test_ai.py]: I introduced these variables only due to Rust quirks, I'd prefer not to have them. You definitely don't need them.

File: yo1k/tic_tac_toe/test/ai/test_ai.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            act_queue_px = DefaultActionQueue(px_id)
+            act_queue_po = DefaultActionQueue(po_id)
+            state = State(
+                    game_rounds=5,
+                    board=Board(),
+                    players=(player_x, player_o))
+            logic = Logic((act_queue_px, act_queue_po))
+            ai_x = RandomAI(player_x.id, ai_rng_seed_px, act_queue_px)
+            ai_o = RandomAI(player_o.id, ai_rng_seed_po, act_queue_po)
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/test/ai/test_ai.py]: You don't need these variables.

File: yo1k/tic_tac_toe/test/ai/test_ai.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            world = World(state, logic, (ai_x, ai_o))
+            possible_round_loops = (state.board.size() ** 2 + 1) * state.game_rounds
+            for _ in range(possible_round_loops):
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/test/ai/test_ai.py]: `possible_round_loops` -> `enough_iterations`?

File: yo1k/tic_tac_toe/ai/ai.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    def act_inround(self, state: State) -> None:
+        if self._player_id != state.turn():
+            return
+        empty_cells_cnt = state.board.size() ** 2 - state.step
+        shift = self._rng.randrange(empty_cells_cnt)
+        for x in range(state.board.size()):
+            for y in range(state.board.size()):
+                if state.board.get(Cell(x, y)) is None:
+                    if shift == 0:
+                        self._action_queue.add(Action.new_occupy(Cell(x, y)))
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/ai/ai.py]: There is no need to instantiate `Cell` twice. This is a place where adding a local variable makes sence.

File: yo1k/tic_tac_toe/ai/ai.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        elif state.phase is Phase.INROUND:
+            self.act_inround(state)
+        else:
+            assert False
+
+    def act_beginning_outround(self, state: State) -> None:
+        if self._player_id in state.required_ready:
+            self._action_queue.add(Action.new_ready())
+
+    def act_inround(self, state: State) -> None:
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/ai/ai.py]: These methods are not supposed to be public.

**Review by stIncMale — CHANGES_REQUESTED** [review]

File: yo1k/tic_tac_toe/test/ai/test_ai.py (hunk 1 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/test/ai/test_ai.py]: Done.

File: yo1k/tic_tac_toe/test/ai/test_ai.py (hunk 2 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/test/ai/test_ai.py]: Done.

File: yo1k/tic_tac_toe/test/ai/test_ai.py (hunk 3 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/test/ai/test_ai.py]: Done.

File: yo1k/tic_tac_toe/ai/ai.py (hunk 4 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/ai/ai.py]: Done.

File: yo1k/tic_tac_toe/ai/ai.py (hunk 5 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/ai/ai.py]: Done.

**Review by stIncMale — APPROVED** [review]

**Pull request merged**